### PR TITLE
parser, expression: support SEPARATOR in group_concat aggregate function

### LINKED ISSUE #5411: group_concat separator not support.
```
mysql> create table test(id int, name int);
Query OK, 0 rows affected (0.11 sec)

mysql> insert into test values(1, 10);
Query OK, 1 row affected (0.00 sec)

mysql> insert into test values(1, 20);
Query OK, 1 row affected (0.00 sec)

mysql> insert into test values(1, 30);
Query OK, 1 row affected (0.00 sec)

mysql> insert into test values(2, 20);
Query OK, 1 row affected (0.00 sec)

mysql> insert into test values(3, 200);
Query OK, 1 row affected (0.00 sec)

mysql> insert into test values(3, 500);
Query OK, 1 row affected (0.00 sec)

mysql> select id, group_concat(name) from test group by id;
+------+--------------------+
| id   | group_concat(name) |
+------+--------------------+
|    1 | 10,20,30           |
|    2 | 20                 |
|    3 | 200,500            |
+------+--------------------+
3 rows in set (0.00 sec)

mysql> select id, group_concat(name separator ';') from test group by id;
ERROR 1105 (HY000): line 1 column 38 near " ';') from test group by id" (total length 65)

mysql> select tidb_version();
+--------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                               |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v1.1.0-alpha-268-gd6b69ad7
Git Commit Hash: d6b69ad7eee3956da8dd6a9ae40686ee43c6ad08
Git Branch: master
UTC Build Time: 2017-12-14 11:49:30 |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by winkyao**
fix #5411 

@XuHuaiyu @shenli PTAL

### CODE REVIEW

File: expression/aggregation/concat.go [hunk 1]
```diff
@@ -74,8 +95,8 @@ func (cf *concatFunction) Update(ctx *AggEvaluateContext, sc *stmtctx.StatementC
 	if ctx.Buffer == nil {
 		ctx.Buffer = &bytes.Buffer{}
 	} else {
-		// now use comma separator
-		ctx.Buffer.WriteString(",")
+		// now use separator
```
Comment (XuHuaiyu) [inline expression/aggregation/concat.go]: this comment is useless

File: parser/parser.y [hunk 2]
```diff
@@ -3445,6 +3449,16 @@ SumExpr:
 		$$ = &ast.AggregateFuncExpr{F: $1, Args: []ast.ExprNode{$4}, Distinct: $3.(bool)}
 	}
 
+OptGConcatSeparator:
+        {
+            	$$ = ast.NewValueExpr(",")
+        }
+| "SEPARATOR" stringLit
+	{ 
+		$$ = ast.NewValueExpr($2);
```
Comment (XuHuaiyu) [inline parser/parser.y]: `;` useless?

File: parser/parser.y [hunk 3]
```diff
@@ -3428,12 +3430,14 @@ SumExpr:
...
+|	builtinGroupConcat '(' BuggyDefaultFalseDistinctOpt ExpressionList OptGConcatSeparator ')'
 	{
-		$$ = &ast.AggregateFuncExpr{F: $1, Args: $4.([]ast.ExprNode), Distinct: $3.(bool)}
+		args := $4.([]ast.ExprNode)
+		args = append(args, $5.(ast.ExprNode))
+		$$ = &ast.AggregateFuncExpr{F: $1, Args: args, Distinct: $3.(bool)}
 	}
 |	builtinMax '(' BuggyDefaultFalseDistinctOpt Expression ')'
-	{
+	{
```
Comment (coocood) [inline parser/parser.y]: remove this modification.

**Comment by winkyao** [comment]
@coocood @XuHuaiyu PTAL

**Comment by XuHuaiyu** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**